gr-uhd/lib/rfnoc_rx_streamer_impl.cc

```diff
@@ -178,7 +179,10 @@ void rfnoc_rx_streamer_impl::flush()
     }
 
     const size_t itemsize = output_signature()->sizeof_stream_item(0);
-    while (true) {
+    // If we don't get an error, time out after 2 seconds
+    gr::high_res_timer_type end_time =
+        gr::high_res_timer_now() + gr::high_res_timer_tps() * 2;
+    while (gr::high_res_timer_now() < end_time) {
         d_streamer->recv(outputs, nbytes / itemsize / d_vlen, d_metadata, 0.0);
         if (d_metadata.error_code != ::uhd::rx_metadata_t::ERROR_CODE_NONE) {
             break;
```

Comment:
```suggestion
            return;
```

Can you please add a log-warning after the `while` loop? I think this change is good, but I also think it's not expected/normal so it could help to notify people this is happening.